### PR TITLE
Usage guide edit

### PR DESCRIPTION
**Pull request by jeromefv**
## PR Checklist

- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [x] New features are documented, with examples if plot related.
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [x] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [x] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: tutorials/introductory/usage.py [hunk 1]
```diff
@@ -121,8 +121,8 @@
 # =====================================
 #
 # All of plotting functions expect `numpy.array` or `numpy.ma.masked_array` as
-# input.  Classes that are 'array-like' such as `pandas` data objects
-# and `numpy.matrix` may or may not work as intended.  It is best to
+# input.  Classes that are similar to arrays such as `pandas` data objects
+# and `numpy.matrix` may not work as intended.  Common convention is to
```
Comment (timhoffm) [inline tutorials/introductory/usage.py:125]: You can remove ` numpy.matrix`. That class is not recommended anymore by numpy itself, and rarely used in practice nowadays. - Also the matrix example code below.

File: tutorials/introductory/usage.py [hunk 2]
```diff
@@ -149,7 +149,7 @@
 #
 # So one can do (OO-style)
 
-x = np.linspace(0, 2, 100)
+x = np.linspace(0, 2, 100)  # This is data for plotting.
```
Comment (timhoffm) [inline tutorials/introductory/usage.py:152]:
```suggestion
x = np.linspace(0, 2, 100)  # sample data
```
Is this sufficient? Alternatively, "sample data for plotting"?

File: tutorials/introductory/usage.py [hunk 3]
```diff
@@ -164,7 +164,7 @@
 ###############################################################################
 # or (pyplot-style)
 
-x = np.linspace(0, 2, 100)
+x = np.linspace(0, 2, 100)  # This is the same data from the previous plot.
```
Comment (timhoffm) [inline tutorials/introductory/usage.py:167]: Use the same comment as above. "same data" is not particularly important, and it breaks the symmetry.

File: tutorials/introductory/usage.py [hunk 4]
```diff
@@ -203,8 +203,8 @@
 #    nowadays and deprecated. It is only mentioned here because you may still
 #    encounter it in the wild.
 #
-# Typically one finds oneself making the same plots over and over
-# again, but with different data sets, which leads to needing to write
+# Typically users may need to make the same plots over and over
```
Comment (timhoffm) [inline tutorials/introductory/usage.py:206]: "Typically" may be a slight over-statement.



```suggestion
# If you need to make the same plots over and over again, but
# with different data sets, it is recommended to write custom
# plotting functions.
```

File: tutorials/introductory/usage.py [hunk 5]
```diff
@@ -290,17 +290,17 @@ def my_plotter(ax, data1, data2, param_dict):
...
-# 3. The function :func:`matplotlib.use`
+# - The :rc:`backend` parameter in your :file:`matplotlibrc` file
+# - The :envvar:`MPLBACKEND` environment variable
+# - The function :func:`matplotlib.use`
 #
-# A more detailed description is given below.
+# Below is a more detailed description.
 #
-# If multiple of these are configurations are present, the last one from the
+# If there are more than one configurations present, the last one from the
```
Comment (timhoffm) [inline tutorials/introductory/usage.py:299]:
```suggestion
# If there is more than one configuration present, the last one from the
```

File: tutorials/introductory/usage.py [hunk 6]
```diff
@@ -97,8 +97,8 @@
 # :class:`~matplotlib.axis.Axis`
 # ------------------------------
 #
-# These are the number-line-like objects. They take
-# care of setting the graph limits and generating the ticks (the marks
+# These are the objects most similar to a number line.
+# They place graph limits and generate ticks (the marks
```
Comment (story645) [inline tutorials/introductory/usage.py:101]: why is "place graph limits" clearer than "set graph limits"?

File: tutorials/introductory/usage.py [hunk 7]
```diff
@@ -121,8 +121,8 @@
 # =====================================
 #
 # All of plotting functions expect `numpy.array` or `numpy.ma.masked_array` as
-# input.  Classes that are 'array-like' such as `pandas` data objects
-# and `numpy.matrix` may or may not work as intended.  It is best to
+# input.  Classes that are similar to arrays such as `pandas` data objects
```
Comment (story645) [inline tutorials/introductory/usage.py:124]: you may want to include (`array-like`) in parenthesis as that's the numpydoc standard and so shows up all over the doc strings.

File: tutorials/introductory/usage.py (hunk 2 above)
Comment (story645) [inline tutorials/introductory/usage.py:152]: what about # 100 values, equally spaced between 0 & 2

File: tutorials/introductory/usage.py (hunk 3 above)
Comment (story645) [inline tutorials/introductory/usage.py:167]: how does this comment help give context?

File: tutorials/introductory/usage.py [hunk 8]
```diff
@@ -243,15 +243,15 @@ def my_plotter(ax, data1, data2, param_dict):
...
+# or if you wanted to have two sub-plots:
 
 fig, (ax1, ax2) = plt.subplots(1, 2)
 my_plotter(ax1, data1, data2, {'marker': 'x'})
 my_plotter(ax2, data3, data4, {'marker': 'o'})
 
 ###############################################################################
-# For these simple examples this style seems like overkill, however
-# once the graphs get slightly more complex it pays off.
+# For these simple examples, this may seem over-engineered; however,
```
Comment (story645) [inline tutorials/introductory/usage.py:253]: I'm a big fan of the "no just/no simple" doc writing advice - "For these examples, this"

File: tutorials/introductory/usage.py [hunk 9]
```diff
@@ -376,11 +376,11 @@ def my_plotter(ax, data1, data2, param_dict):
 # ``macosx`` backends.  An alternative renderer is based on the Cairo library,
 # used by ``Qt5Cairo``, ``Qt4Cairo``, etc.
 #
-# For the rendering engines, one can also distinguish between `vector
+# For the rendering engines, users can also distinguish between `vector
 # <https://en.wikipedia.org/wiki/Vector_graphics>`_ or `raster
 # <https://en.wikipedia.org/wiki/Raster_graphics>`_ renderers.  Vector
 # graphics languages issue drawing commands like "draw a line from this
-# point to this point" and hence are scale free, and raster backends
+# point to this point" and hence are scale free. Raster backends
```
Comment (story645) [inline tutorials/introductory/usage.py:383]: I think raster should be lowercase

File: tutorials/introductory/usage.py [hunk 10]
```diff
@@ -439,7 +439,7 @@ def my_plotter(ax, data1, data2, param_dict):
 # ========= ================================================================
 #
 # .. note::
-#    The names of builtin backends case-insensitive; e.g., 'Qt5Agg' and
+#    The names of builtin backends case-insensitive. For example, 'Qt5Agg' and
```
Comment (story645) [inline tutorials/introductory/usage.py:442]: nit/typo: "The names of builtin backends are case-insensitive."

File: tutorials/introductory/usage.py [hunk 11]
```diff
@@ -657,7 +657,7 @@ def my_plotter(ax, data1, data2, param_dict):
 # :rc:`path.simplify` and :rc:`path.simplify_threshold`, which
 # can be defined e.g. in the :file:`matplotlibrc` file (see
 # :doc:`/tutorials/introductory/customizing` for more information about
-# the :file:`matplotlibrc` file). :rc:`path.simplify` is a boolean
+# the :file:`matplotlibrc` file). :rc:`path.simplify` is a Boolean
```
Comment (story645) [inline tutorials/introductory/usage.py:660]: I think boolean is lowercase?

File: tutorials/introductory/usage.py [hunk 12]
```diff
@@ -685,14 +685,14 @@ def my_plotter(ax, data1, data2, param_dict):
...
-# to use a different value, you can change your :file:`matplotlibrc`
-# file.  Alternatively, you could create a new style for
-# interactive plotting (with maximal simplification) and another
-# style for publication quality plotting (with minimal
-# simplification) and activate them as necessary. See
-# :doc:`/tutorials/introductory/customizing` for
-# instructions on how to perform these actions.
+# threshold of ``1/9``. To change default settings to use a different
+# value, change the :file:`matplotlibrc` file. Alternatively, users
+# can create a new style for interactive plotting (with maxmial
```
Comment (story645) [inline tutorials/introductory/usage.py:690]: typo? maxmial

**Review by story645 — CHANGES_REQUESTED** [review]
mostly small things/nits but I like the changes here.

File: tutorials/introductory/usage.py (hunk 2 above)
Comment (timhoffm) [reply tutorials/introductory/usage.py:152]: IMHO that'd be one level too low. It describes the actual content of x, which is (more or less clearly) already stated by the code. But more importantly, the exact content is not particularly relevant. If we want a comment, it should state the purpose, hence my proposal above "sample data".

File: tutorials/introductory/usage.py (hunk 1 above)
Comment (tacaswell) [reply tutorials/introductory/usage.py:125]: I think this is a bit too strong, we do test that pandas works correctly in some cases and special case extracting the index and label when it makes sense.

File: tutorials/introductory/usage.py (hunk 2 above)
Comment (tacaswell) [reply tutorials/introductory/usage.py:152]: I agree with Tim here.

File: tutorials/introductory/usage.py (hunk 9 above)
Comment (tacaswell) [reply tutorials/introductory/usage.py:383]: It starts a sentence?

**Review by tacaswell — APPROVED** [review]
I am :+1: on these changes.  I agree with many of Tim's suggestions, but in the spirit of "does this makes the docs better" I would be fine with this going in either way.

**Comment by tacaswell** [comment]
@jeromefv Ping me if you want a hand with the rebase.

**Comment by dstansby** [comment]
👋 @jeromefv are you still planning on finishing this? I think it just needs a rebase. If not, I'm happy to finish it off instead.

**Comment by jeromefv** [comment]
Thank you, @dstansby! I'll work on it today and rebase with the changes. I'll ping any folks if I run into issues. Thank you for the update!

File: tutorials/introductory/usage.py (hunk 3 above)
Comment (jeromefv) [reply tutorials/introductory/usage.py:167]: I followed @timhoffm's suggestions and have "Sample data." as the comment here instead.

File: tutorials/introductory/usage.py (hunk 8 above)
Comment (jeromefv) [reply tutorials/introductory/usage.py:253]: Reduced the language even more now!

File: tutorials/introductory/usage.py (hunk 11 above)
Comment (jeromefv) [reply tutorials/introductory/usage.py:660]: Found some inconsistencies in the Python docs, [Build-in Types](https://docs.python.org/3/library/stdtypes.html#boolean-operations-and-or-not) use the capital "B" and [Boolean Objects](https://docs.python.org/3/c-api/bool.html?highlight=boolean) does not.

**Comment by jeromefv** [comment]
Overall, a revamp of this document would fit more in line with how I'd like to create consistency between this and the `Getting Started` guide in my other PR. I think it would take quite a bit of time to approach this again with much more in-depth rewriting.

**Comment by story645** [comment]
@jeromefv just to clarify, does that mean you'll finish out this PR but think more or needs to be done or should @dstansby finish it out?

**Comment by jeromefv** [comment]
> @jeromefv just to clarify, does that mean you'll finish out this PR but think more or needs to be done or should @dstansby finish it out?

That's a good point! If @dstansby will/wants to take over and do bigger sweeping changes, then it makes a lot of sense for me to sit out here. Originally, this PR was more of a "practice" for me to get an idea of how I can shape the language to better fit the explanations.

I think if I were to tackle this PR again, I'd do a large restructuring project with explanations similar to what GSoD was. I don't think I'd be able to do that in a prompt timeline at the moment, however.

**Review by story645 — APPROVED** [review]

**Pull request merged**